src/storage-controller/src/collection_mgmt.rs

```diff
@@ -400,6 +415,19 @@ where
 
         Ok(MonotonicAppender::new(tx))
     }
+
+    fn get_read_only_rx(&self, id: GlobalId, force_writable: bool) -> watch::Receiver<bool> {
+        if *self.read_only_rx.borrow() && force_writable {
```

Comment:
Why the extra `*self.read_only_rx.borrow()` in the condition? Isn't `force_writable == true` enough?

Reply:
It's technically possible for `force_writable == true` after leadership promotion if we hit the rehydration timeout before migrating the builtin source (which is extremely unlikely) when in read-only mode. In that case I'd just rather use `self.read_only_rx.clone()` instead of `self.hacky_always_false_watch.1.clone()`, even though they're both likely `false`.

However, I just modified how we calculate `force_writable` to consider read only mode so that we can remove this check.